### PR TITLE
Add contains, reverse operations for Var

### PR DESCRIPTION
**Pull request by martinxu9**
## Summary
- Add reverse operation for list type Var.
- Add contains operation for list/tuple/dict/str type Var, overrides `__contains__` to raise since this cannot be used to return a Var at the end, it gets converted to boolean. 
- UTs for reverse and contains.
- Pending: add code snippet/example app
## Tests
- [x] new UTs
## Examples
  - `foreach` renders list of texts in the reverse order
  - text box at the bottom shows a boolean printed as string for whether the user typed input is in the list of words.
```
class State(rx.State):
    """The app state."""
    words: List[str] = ["Hello", "world", "from", "Reflex!"]
    typed_word: str = ""

def index() -> rx.Component:
    return rx.vstack(
        rx.foreach(State.words.reverse(), lambda word: rx.text(word)),
        rx.input(value=State.typed_word, on_change=State.set_typed_word, style=dict(width="10vw")),
        rx.text("Did I type a word in the above?"),
        rx.text(State.words.contains(State.typed_word).to_string()),
    )
```
image
image

### CODE REVIEW

File: reflex/vars.py [hunk 1]
```diff
@@ -685,6 +685,74 @@ def __ror__(self, other: Var) -> Var:
...
+            TypeError: If the var is not a list.
+
+        Returns:
+            A var with the reversed list.
+        """
+        if not types._issubclass(self.type_, list):
+            raise TypeError(f"Cannot reverse non-list var {self.full_name}.")
+
+        return BaseVar(
+            name=f"{self.full_name}.reverse()",
```
Comment (masenf) [inline reflex/vars.py:751]: this gives me hydration errors when i try it in dev mode because it will reverse the state var in place.

```suggestion
            name=f"[...{self.full_name}].reverse()",
```

(render functions are not allowed to mutate values)

**Review by masenf — CHANGES_REQUESTED** [review]

File: reflex/vars.py [hunk 2]
```diff
@@ -685,6 +685,74 @@ def __ror__(self, other: Var) -> Var:
...
         return self.operation("||", other, type_=bool, flip=True)
 
+    def __contains__(self, _: Any) -> Var:
+        """Override the 'in' operator to alert the user that it is not supported.
+
+        Raises:
+            TypeError: the operation is not supported
+        """
+        raise TypeError(
+            "'in' operator not supported for Var types, use Var.contains() instead."
```
[the remark this answers is not in the document]
Comment (martinxu9) [reply reflex/vars.py:695]: so the return of `__contains__` seems to be automatically converted to boolean, so instead of something like a var with `{[1,2,3].includes(1)}`, we get `True` (the var being eval as a boolean).

File: reflex/vars.py [hunk 3]
```diff
@@ -685,6 +685,74 @@ def __ror__(self, other: Var) -> Var:
...
+            raise TypeError(
+                f"Var {self.full_name} of type {self.type_} does not support contains check."
+            )
+        if isinstance(other, str):
+            other = Var.create(json.dumps(other), is_string=True)
+        elif not isinstance(other, Var):
+            other = Var.create(other)
+        if types._issubclass(self.type_, Dict):
+            return BaseVar(
+                name=f"{self.full_name}.has({other.full_name})",
```
[the remark this answers is not in the document]
Comment (martinxu9) [reply reflex/vars.py:720]: Yes did not want the enclosing curly brackets.

File: reflex/vars.py [hunk 4]
```diff
@@ -685,6 +685,74 @@ def __ror__(self, other: Var) -> Var:
...
+        """Reverse a list var.
+
+        Raises:
+            TypeError: If the var is not a list.
+
+        Returns:
+            A var with the reversed list.
+        """
+        if not types._issubclass(self.type_, list):
+            raise TypeError(f"Cannot reverse non-list var {self.full_name}.")
```
[the remark this answers is not in the document]
Comment (martinxu9) [reply reflex/vars.py]: good catch, thanks. Added another place to check None first; also added test with var type_ is None

File: reflex/vars.py (hunk 1 above)
Comment (martinxu9) [reply reflex/vars.py]: 👍🏼

**Comment by martinxu9** [comment]
@picklelo Added some example code

**Comment by martinxu9** [comment]
@masenf new upload incorporating comments and suggestions

**Review by masenf — APPROVED** [review]
